src/test/java/io/supertokens/test/oauth2/StorageTests.java

```diff
@@ -63,27 +67,235 @@ public void testCreateOAuth2ClientTransaction() throws Exception {
 
         OAuth2SQLStorage storage = (OAuth2SQLStorage) StorageLayer.getStorage(process.getProcess());
 
-        List<String> redirectUris = new ArrayList<>();
-        redirectUris.add("r1");
-        redirectUris.add("r2");
+        OAuth2Client oAuth2Client = getOAuth2Client(null, null, null);
+        createOAuth2Client_TransactionHelper(storage ,oAuth2Client);
+
+        OAuth2Client oAuth2ClientFromGetQuery = storage.getOAuth2ClientById(new AppIdentifier(null, null),
+                oAuth2Client.clientId );
+
+        Assert.assertEquals(oAuth2Client, oAuth2ClientFromGetQuery);
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    @Test
+    public void testCreateOAuth2ClientTransactionExceptions() throws Exception {
```

Comment:
i think we should make it so that clients have a unique name for a given app. This will require change to the schema, plus a new exception type to be added to the plugin interface for this function.

So you should also add a new test case in this function for that new function

Reply:
Yes , correct.
However this unique key constraint should be on `app_id` and `name` columns, so that we have an uniquely named client for every app.

Let me know if i am missing something here.

Reply:
Changes made